src/filters/Component.jsx

```diff
@@ -0,0 +1,377 @@
+                filterValue: newValue.toDate().getTime(),
+              });
+            }}
+            openTo="date"
+            style={{width: '100px', cursor: 'pointer'}}
+            value={date}
+            variant="inline"
+          />
+        </MuiPickersUtilsProvider>
+      </ThemeProvider>
+    </StylesProvider>
+  );
+};
+
+const FiltersInputDropDown = ({filterHandler, onFilterChange}) => {
+  return (
+    <SelectDropDown
+      className="input-dropdown"
+      initialValue={filterHandler.getDefaultValue()}
+      items={filterHandler.getDropdownInputItems()}
+      onValueChange={(filterValue) =>
+        onFilterChange({
+          ...filterHandler.toAttrsValue(),
+          filterValue,
+        })
+      }
+    />
+  );
+};
+
+const FiltersTextInput = ({filterHandler, onFilterChange}) => {
+  const inputRef = useRef();
+
+  useEffect(() => {
+    if (inputRef.current) inputRef.current.focus();
+  }, [inputRef]);
+
+  return (
+    <input
+      className="filter-value-input"
+      defaultValue={filterHandler.getDefaultValue()}
+      onChange={(e) =>
+        onFilterChange({
+          ...filterHandler.toAttrsValue(),
+          filterValue: e.target.value,
+        })
+      }
+      ref={inputRef}
+    ></input>
+  );
+};
+
+const FiltersLabelsDropDown = ({filterHandler, onFilterChange}) => {
+  const [openDropDown, setOpenDropDown] = useState(false);
+
+  const tablePos = tableFiltersMenuKey.getState(filterHandler.view.state);
+  const handleLabelChoose = (title, checked, allChosenLabels) => {
+    let newChosenLabels = allChosenLabels.map((label) => label.title);
+
+    if (checked) {
+      newChosenLabels = [...newChosenLabels, title];
+    } else {
+      newChosenLabels = newChosenLabels.filter((label) => label !== title);
+    }
+
+    onFilterChange({
+      ...filterHandler.toAttrsValue(),
+      filterValue: newChosenLabels,
+    });
+  };
+
+  return (
+    <div className="selectDropDownContainer">
+      <SelectDropDownButton
+        disableDropDown={false}
+        itemStyleClass={''}
+        label={'Choose Labels'}
+        openDropDown={() => setOpenDropDown(!openDropDown)}
+      />
+      {openDropDown && (
+        <LabelsChooser
+          handleLabelChoose={handleLabelChoose}
+          inFilters={true}
+          initialChosenLabels={filterHandler
+            .toAttrsValue()
+            .filterValue.map((label) => ({title: label}))}
+          node={filterHandler.table}
+          onClose={() => setOpenDropDown(false)}
+          pos={tablePos.pos + 1}
+          view={filterHandler.view}
+        />
+      )}
+    </div>
+  );
+};
+
+const getInputElement = (filterHandler) => {
+  const inputType = filterHandler.getInputType();
+
+  switch (inputType) {
+    default:
+      return () => <></>;
+    case 'input':
+      return FiltersTextInput;
+    case 'date-picker':
+      return FiltersDatePicker;
+    case 'dropdown':
+      return FiltersInputDropDown;
+    case 'labels-dropdown':
```

Comment:
do we want an enum for this? maybe could be shared with `types.config`?

Reply:
were still working with js here so enum is not an option for now. might be nice improvement to make it a ts repo :)  
we can make some kind of map if it makes more sense and feels more natural

Reply:
sounds good! can add a ticket if you see necessary